frontend/common/src/messages/apiMessages.ts

```diff
@@ -5,55 +5,150 @@ import { defineMessages, MessageDescriptor } from "react-intl";
+    id: "6O1sjV",
+    description:
+      "Error message that the given user identifier is already in use when updating.",
+  },
+  EmailAddressInUse: {
+    defaultMessage: "Cannot update - this email address is already in use.",
+    id: "VqrVpT",
+    description:
+      "Error message that the given email address is already in use when updating.",
+  },
+  CreateUserSubInUse: {
+    defaultMessage:
+      "Cannot create - this user identifier (sub) is already in use",
+    id: "CVmEkf",
     description:
-      "Error message that the given user identifier is already in use.",
+      "Error message that the given user identifier is already in use when creating.",
   },
-  "unique:users,email": {
-    defaultMessage: "This email address is already in use",
-    id: "nDaEgr",
+  CreateUserEmailInUse: {
+    defaultMessage: "Cannot create - this email address is already in use.",
+    id: "WUAAr1",
     description:
-      "Error message that the given email address is already in use.",
+      "Error message that the given user identifier is already in use when creating.",
   },
-  "unique:skill_families,key": {
+
+  // skill validation
+  SkillFamilyKeyStringInUse: {
     defaultMessage: "This skill family key string is already in use",
     id: "XTuwjA",
     description:
       "Error message that the given skill family key is already in use.",
   },
-  "validator:archived_at.prohibited": {
-    defaultMessage: "already archived",
-    id: "OtOIXG",
-    description: "Error message that the given model is already archived.",
+
+  // application validation
+  AlreadyArchived: {
+    defaultMessage: "Application is already archived.",
+    id: "A+dFlE",
+    description:
+      "Error message that the given application is already archived.",
+  },
+  AlreadySubmitted: {
+    defaultMessage: "Application was already submitted.",
+    id: "x/wHOH",
+    description:
+      "Error message that the given application is already submitted.",
+  },
+  "pool candidates status InvalidValueArchival": {
+    defaultMessage: "This application may not yet be archived.",
+    id: "TITln/",
+    description: "Error message that the application cannot be archived.",
   },
+  "pool candidates status InvalidValueDeletion": {
+    defaultMessage: "This application may not be deleted.",
+    id: "YUM0Mo",
+    description: "Error message that the application cannot be deleted.",
+  },
+
   "validator:expiry_date.after": {
     defaultMessage: "Expiry Date must be after today.",
     id: "sfr5Pa",
     description:
       "Error message that the given skill expiry date must be after today.",
   },
-  "validator:advertisement_location.*.required_if": {
+
+  // pool updating
+  UpdatePoolExpiryDate: {
+    defaultMessage: "The pool must have an expiry date after today.",
+    id: "gU/2O6",
+    description: "Error message that pool expiry isn't in the future.",
+  },
+
+  // pool publishing validation
```

Comment:
updated messages here https://github.com/GCTC-NTGC/gc-digital-talent/pull/4106/commits/281c8f09b161bc4fbb59d0840c4d5e45662346f0
some merited unique messages I felt, like essential skills and how we've got location working, they need to be more explicit to avoid confusion as to what needs to be done

Reply:
thanks @Jerryescandon